src/server/i18n/index.js

```diff
@@ -17,12 +17,28 @@
  * under the License.
  */
 
+import { resolve } from 'path';
+import globby from 'globby';
 import { i18n, i18nLoader } from '@kbn/i18n';
 
 export async function i18nMixin(kbnServer, server, config) {
-  const { translationPaths = [] } = kbnServer.uiExports;
   const locale = config.get('i18n.locale');
 
+  const groupedEntries = await Promise.all(
+    config.get('i18n.translationsScanDirs').map(async path => {
```

Comment:
Sorry for asking for more changes here, but I don't remember seeing some of this config last time I looked. I think this implementation offers too many possible options, like `src/ui/foo/plugins/bar/translations/en.json`... That doesn't really make sense to support and potentially confusing if it conflicts with something we don't expect today.

Lets remove the `i18n.translationsScanDirs` config, and automatically scan the `plugins.scanDirs` and `plugins.paths` configs without copying them to a new config options. Also, the paths from those configs and `src/ui` should all be searched differently:

- `plugins.scanDirs` -> `globby('*/translations/*.json', { cwd })` only match translations within children of the scan dir
- `plugins.paths` -> `globby('translations/*.json', { cwd })` only match translations within each specific plugin
- `fromRoot('src/ui/translations')` -> `globby('*.json', { cwd })` only match translations in `src/ui/translations`